src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -100,26 +59,146 @@ class ReactShallowRenderer {
       element.type,
     );
 
-    // TODO Remove this wrapper if/when context parameter is removed
-    if (context && Object.keys(context).length) {
-      element = wrapElementWithContextProvider(element, context);
-    }
+    this._element = element;
 
-    this._renderer = ReactTestRenderer.create(element, {
-      createNodeMock: createShallowNodeMock(),
-    });
+    if (this._instance) {
+      this._rendered = updateClassComponent(
+        this._instance,
+        element.props,
+        context,
+      );
+    } else {
+      const prototype = element.type.prototype;
+
+      if (typeof prototype.render === 'function') {
```

Comment:
Those places do it the way you mentioned- which is great, I'm happy to change to that. 😁 

There's a few places we're check for a `render` function too but they're probably just older:
* [`ReactFiberBeginWork`](https://github.com/facebook/react/blob/10eaf26809bdb1874e14f0879190c356937c2e3e/src/renderers/shared/fiber/ReactFiberBeginWork.js#L517-L519)
* [`reactComponentExpect`](https://github.com/facebook/react/blob/10eaf26809bdb1874e14f0879190c356937c2e3e/src/test/reactComponentExpect.js#L189-L190)
* [`findNodeHandle`](https://github.com/facebook/react/blob/10eaf26809bdb1874e14f0879190c356937c2e3e/src/renderers/native/findNodeHandle.js#L103)
* [`findDOMNode`](https://github.com/facebook/react/blob/10eaf26809bdb1874e14f0879190c356937c2e3e/src/renderers/dom/shared/findDOMNode.js#L70)